### PR TITLE
UNR-407: Adding FUnrealObjectRef assignment when sending/receiving updates

### PR DESCRIPTION
**Pull request by danielimprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR adds the assignment of the `FUnrealObjectRef` contexts for non-arrays. It also updates the GDK to use the `FUnrealObjectRef` from [this engine change](https://github.com/improbableio/UnrealEngine/pull/14).

Blueprint support is still not present but a new ticket has been added to track this. [UNR-635](https://improbableio.atlassian.net/browse/UNR-635)

#### Tests
This PR was validated with a test in the following [TestSuite PR](https://github.com/spatialos/UnrealGDKTestSuite/pull/110)

#### Documentation
Related [Jira ticket](https://improbableio.atlassian.net/browse/UNR-407)

No documenation provided.

#### Primary reviewers
@Vatyx @improbable-valentyn

### CODE REVIEW

**Comment by Vatyx** [comment]
Is this change needed so we can access it in Blueprints?

Also @danielimprobable could you also put the name of the ticket in the PR instead of just the number? I always have to look up the ticket when I see the number.

**Comment by danielimprobable** [comment]
@Vatyx This is needed so that we can generate the object refs for all classes, The PR is work in progress and therefore I haven't spent more time updateing the details yet. Will update later today

File: SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #include "Interop/Connection/SpatialWorkerConnection.h"
 #include "Interop/SpatialReceiver.h"
 #include "Interop/SpatialSender.h"
-#include "Schema/UnrealObjectRef.h"
+#include "UnrealObjectRef.h"
```
Comment (Vatyx) [inline SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp]: I think we should keep UnrealObjectRef.h in the Schema folder to indicate this type is defined in schema.

Comment (danielimprobable) [reply SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp]: It has moved to the engine, so I didn't add one in there, but I guess I could add a folder indirection there instead?

Comment (Vatyx) [reply SpatialGDK/Source/Private/EngineClasses/SpatialPackageMapClient.cpp]: Ah ok I just saw that it had been moved to the Engine. Yeah I think if there was something that made it really obvious this is GDK code like `"improbable/UnrealObjectRef.h"` that would be best.

File: SpatialGDK/Source/Private/Utils/ComponentFactory.cpp [hunk 2]
```diff
@@ -388,4 +395,19 @@ Worker_ComponentUpdate ComponentFactory::CreateHandoverComponentUpdate(Worker_Co
 	return ComponentUpdate;
 }
 
+void ComponentFactory::AssignUnrealObjectRefToContext(UProperty* Property, const uint8* Data, FUnrealObjectRef ObjectRef)
```
Comment (Vatyx) [inline SpatialGDK/Source/Private/Utils/ComponentFactory.cpp]: `AssignUnrealObjectRefToSpatialOSContext`? Or too long?

File: SpatialGDK/Source/Public/EngineClasses/SpatialNetBitReader.h [hunk 3]
```diff
@@ -5,14 +5,14 @@
 #include "CoreMinimal.h"
 #include "UObject/CoreNet.h"
 
-#include "Schema/UnrealObjectRef.h"
+#include "improbable/UnrealObjectRef.h"
```
Comment (Vatyx) [inline SpatialGDK/Source/Public/EngineClasses/SpatialNetBitReader.h]: Nice

**Pull request merged**